Simple Incremental/Assets/Scripts/EnemyTargeting.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyTargeting : MonoBehaviour
+{
+    // All enemy damage types are added here
+    [SerializeField]
+    List<EnemyDamage> enemyDamage = new List<EnemyDamage>(); 
+
+    // List of gameObjects tagged as player
+    List<GameObject> playerTargets = new List<GameObject>(); 
+
+    // List of player objects and best attack for each
+    List <PlayerTarget> playerTargetList= new List<PlayerTarget>();
+
+    bool targetListCurrent = false;
+    PlayerTarget bestTarget;
+    Vector3[] movement = new Vector3[2];
+
+    // Delay for checking movement to avoid checks on every frame
+    float movementCheckDelay = 1.0f;
+    float elapsedTime;
+
+
+    private void Update()
+    {
+        elapsedTime += Time.deltaTime;
+
+        if (elapsedTime >= movementCheckDelay)
+        {
+            MovementCheck();
+            elapsedTime = 0;
+        }
+    }
+
+
+    public PlayerTarget FindATarget()
+    {
+        // Don't recalculate if the list is updated
+        if (targetListCurrent)
+        {
+            return bestTarget;
+        }
+        else
+        {
+            PopulateTargetList();
+
+            // Record current enemy position
+            movement[0] = transform.position;
+            return bestTarget;

```

Comment:
In Either case your returning best target so you could simplify your code. Like the following.

if(!targetListCurrent)
{
   PopulateTargetList();
   movement[0] = transform.position;
}
return bestTarget;